httputil/httputil.go

```
@@ -153,8 +183,22 @@ func DownloadBinary(originURL, destDir, destFile string) (string, error) {
 			}
 		}()
 
+		u, err := url.Parse(originURL)
+		if err != nil {
+			// originURL supposed to be valid
+			panic(err)
```

Please return an error instead